examples/dreambooth/train_dreambooth.py

```diff
@@ -351,7 +351,18 @@ def main():
                 images = pipeline(example["prompt"]).images
 
                 for i, image in enumerate(images):
-                    image.save(class_images_dir / f"{example['index'][i] + cur_class_images}.jpg")
+                    image_filename = f"{example['index'][i] + cur_class_images}"
+
+                    # Add and underscore at the end of the filename until there is no file with the same name
+                    while True:
+                        image_path = class_images_dir / (image_filename + ".jpg")
+
+                        if not os.path.exists(image_path):
+                            break
+
+                        image_filename = f"{image_filename}_"
+
+                    image.save(image_path)
```

Comment:
Think we can just add a timestamp to the `image_filename` so each name will be unique every time.
```suggestion
                    timestr = time.strftime("%Y%m%d-%H%M%S")
                    image_filename = f"(class_images_di / {example['index'][i] + cur_class_images}-{timestr}"
                    image.save(image_filename)
```

Reply:
Yeah, the logic was a bit weird because there is no do-while loop in Python.
Your suggestion might fail on some very fast computers generating more than one image per second and instead of adding milliseconds, I propose we use the hash of the image, it's cleaner.